docs/fastdds/dds_layer/subscriber/dataReader/readingData.rst

```diff
@@ -9,31 +9,84 @@ Accessing received data
-   so it is no longer accessible.
-
-If there is no unread data in the DataReader, both operations will return
-``NO_DATA`` and nothing is returned.
-
-In addition to the data value, the data access operations also provide a SampleInfo
-instance with additional information that help interpreting the returned data value, like the originating
+ * **Reading** is done with any of the following member functions:
+
+   * |DataReader::read_next_sample-api| reads the next, non-previously accessed data value available
+     on the DataReader, and stores it in the provided data buffer.
+   * |DataReader::read-api|, |DataReader::read_instance-api| and |DataReader::read_next_instance-api|
+     provide mechanisms to get a collection of samples matching certain conditions.
+
+ * **Taking** is done with any of the following member functions:
+
+   * |DataReader::take_next_sample-api| reads the next, non-previously accessed data value available on the DataReader,
+     and stores it in the provided data buffer.
+   * |DataReader::take-api|, |DataReader::take_instance-api| and |DataReader::take_next_instance-api|
+     provide mechanisms to get a collection of samples matching certain conditions.
+
+   When taking data, the returned samples are also removed from the DataReader, so they are no longer accessible.
+
+When there is no data in the DataReader matching the required conditions, all the operations will return
+``NO_DATA`` and output parameter will remain unchanged.
+
+In addition to the data values, the data access operations also provide SampleInfo instances with additional
+information that help interpreting the returned data values, like the originating
 :ref:`dds_layer_publisher_dataWriter` or the publication time stamp.
 Please, refer to the :ref:`dds_layer_subscriber_sampleInfo` section for an extensive description of its contents.
 
+.. _dds_layer_subscriber_accessreceived_loans:
+
+Loaning and Returning Data and SampleInfo Sequences
+---------------------------------------------------
+
+The |DataReader::read-api| and |DataReader::take-api| operations (and their variants) return information to the
+application in two sequences:
+
+ * Received DDS data samples in a sequence of the data type
+ * Corresponding information about each DDS sample in a SampleInfo sequence
+
+These sequences are parameters that are passed by the application code into the
+|DataReader::read-api| and |DataReader::take-api| operations.
+When empty sequences (sequences that are initialized but have a maximum length of 0) are used, the middleware will
+fill those sequences with memory directly loaned from the receive queue itself.
+There is no copying of the data or SampleInfo when the contents of the sequences are loaned.
+This is certainly the most efficient way for the application code to retrieve the data.
+
+When doing so, however, the code must return the loaned sequences back to the middleware so that they can be reused
+by the receive queue.
```

Comment:
```suggestion
When doing so, however, the code must return the loaned sequences back to the middleware, so that they can be reused
by the receive queue.
```